packages/volto/src/components/manage/UniversalLink/UniversalLink.tsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import { HashLink as Link } from 'react-router-hash-link';
+import { useSelector } from 'react-redux';
+import {
+  flattenToAppURL,
+  isInternalURL,
+  URLUtils,
+} from '@plone/volto/helpers/Url/Url';
+import config from '@plone/volto/registry';
+import cx from 'classnames';
+
+interface UniversalLinkProps {
+  href: string;
+  item?: {
+    '@id': string;
+    remoteUrl?: string;
+    getRemoteUrl?: string;
+    '@type'?: string;
+  };
+  openLinkInNewTab?: boolean;
+  download?: boolean;
+  children: React.ReactNode;
+  className?: string;
+  title?: string;
+  smooth?: boolean;
+  onClick?: (e: React.MouseEvent) => void;
+  onKeyDown?: (e: React.KeyboardEvent) => void;
+}
+
+export interface AppState {
+  content: {
+    data?: {
+      '@components'?: {
+        translations?: {
+          items?: { language: string; '@id': string }[];
+        };
+      };
+    };
+  };
+  navroot: {
+    data: {
+      navroot: {
+        id: string;
+      };
+    };
+  };
+  intl: {
+    locale: string;
+  };
+  userSession: {
+    token: string;
+  };
+}
+
+const UniversalLink = React.memo<UniversalLinkProps>(
```

Comment:
What is the reasoning behind the `React.memo`?

Reply:
It's a performance optimization.

The component gets memoized and if we have for example 50 links in a list and only one changes, with React.memo only the changed link get rendered new.

We can use it here because UniversalLink changes only over props, we have no local states or side effects.

Reply:
I made some tests and negative tests to show this optimized behavior and also show when it's not working (unstable props). We always have to use stable references if we want the optimized behavior, that means not passing inline objects to props or inline jsx to children. 

I made some detailed examples in the tests, maybe we should also document this.